org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,300 @@
+                    });
+            promises.push(statePromise);
+        }
+
+        if (Object.keys(nonStatusChanges).length > 0) {
+            var nonStatePromise = this._makeRequest(this._devicesPath + '/' + deviceId, 'GET')
+                    .then((details) => {
+
+                        for (var item in nonStatusChanges) {
+                            details[item] = nonStatusChanges[item];
+                        }
+
+                        return this._makeRequest(this._devicesPath + '/' + deviceId, 'PUT', JSON.stringify(details))
+                                .then(function (response) {
+                                    return Promise.resolve(response);
+                                });
+                    });
+            promises.push(nonStatePromise);
+        }
+
+
+        // Merge Responses
+        var device = {};
+        return Promise.all(promises).then(function (responses) {
+            
+            for (var i = 0; i < responses.length ; i++) {
+                if ( responses[i] !== undefined && responses[i].status == 'succeeded') {
+                    if (responses[i].command !== undefined)
+                    {
+                        if( responses[i].command.level !== undefined) {
+                            device['Level'] = responses[i].command.level;
+                        }
+
+                        if (responses[i].command !== undefined && responses[i].command.command !== undefined) {
+                            device['Power'] = responses[i].command.command;
+                        }
+                    }
+
+                    if (responses[i].content !== undefined) {
+                        for (var item in responses[i].content) {
+                            device[item] = responses[i].content[item];
+                        }
+                    }
+                }
+            }
+            
+            if(device.length == 0 ) return Promise.resolve(undefined);
+            return Promise.resolve(device);
+        });
+
+    }
+
+    /** 
+     * Given the hub specific device, returns the opent2t schema and translator
+     */
+
+    _getOpent2tInfo(devCat, subCat) {
+        switch (devCat) {
+            case 1:
+                if (this._subCatMap.lightBulbs.indexOf(subCat) >= 0)
+                {
+                    return {
+                        "schema": 'org.opent2t.sample.lamp.superpopular',
+                        "translator": "opent2t-translator-com-insteon-lightbulb"
+                    };
+                }
+                return undefined;
+            default:
+                return undefined;
+        }
+    }
+    
+    _isDeviceStatus(propertyName){
+        switch (propertyName) {
+            case "command":
+            case "level":
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    _getCommandResponse(commandId, count){
+
+        return sleep(1000).then(() => {
```

Comment:
Comments added. Also added more error handling information to the method.

---
In reply to: [84781861](https://github.com/openT2T/translators/pull/107#discussion_r84781861) [](ancestors = 84781861,84631791,82258166)